bin/check-all.sh

```diff
@@ -60,9 +60,9 @@ fi
 if [ -e "${base_dir}/gradlew" ]; then
   gradle_file="${base_dir}/gradlew"
 else
-  gradle_base_file=$(which gradle)
+  gradle_base_file=$(which gradle) && chk_gradle=$? || chk_gradle=$?
```

Comment:
Hey @vjagadish1989 thanks again for checking this out with me.
So on my env (Debian Lenny) the script exited when executing `which gradle` and it didn't make it to the `if` statement. `which` was installed because I didn't get a `command not found` error. 
I have also tried to reproduce the issue on different servers now and everything works as expected i.e. `which` doesn't crash. I guess it was just something wrong with my local env.